Demo: programmable money with ERC-20 smart contracts

edv1n opened the pull request:
# Assignment Proposal

## Title

Demo programmable money with smart contract

## Names and KTH ID

  - Edwin So (edwinso@kth.se)

## Deadline

- Week 3

## Category

- Demo

## Description

Demonstration on how to perform programmable money with ETH's smart contract with ERC-20. The presentation will go deep into the mechanism

**Relevance**

Programmable money is one of the major use cases of smart contracts. This demo will demonstrate how common "cryptocurrency coins" are being created and how they work in ETH.

edv1n committed:
Create README.md for assignment proposal - demo

Added a README.md for the assignment proposal detailing the demo of programmable money with smart contracts.

javierron commented:
Thanks for the proposal @edv1n. This is a very relevant topic for the week.

> Demonstration on how to perform programmable money with ETH's smart contract (e.g., with ERC-20). 

Do I understand correctly that you will show how to create/deploy an ERC-20 smart contract?

> Will describe the mechanism, the difference between ETH's "currency" (i.e., gas / ETH)

This is good content for the intro of the demo, however I recommend that the demo is focused in a single topic, to be explored in depth, e.g. ERC-20.

edv1n commented:
@javierron I am planning to focus on one type of contract, and it will very likely be ERC-20. I am now ensuring I can deploy and interact with an ERC-20 contract for the demo.

edv1n committed:
Clarify demo description for ERC-20 smart contracts

Updated the README to clarify the demo's focus on ERC-20 smart contracts and the mechanisms involved.

edv1n commented:
Hi @javierron 

I have updated the description, now the demo will go deep into the mechanism.
